lib/Runtime/PlatformAgnostic/Platform/Linux/HiResTimer.cpp

```diff
@@ -0,0 +1,80 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Runtime.h"
+#include <time.h>
+#include <sys/time.h>
+#include "ChakraPlatform.h"
+
+namespace Js
+{
+    double HiResTimer::GetSystemTime()
+    {
+// todo-osx: mach_absolute_time
+#ifndef __APPLE__
+        struct timespec fast_time;
+        if (clock_gettime(CLOCK_REALTIME, &fast_time) == 0)
+        {
+            return (fast_time.tv_sec * 1000) + (int32_t)(fast_time.tv_nsec/1e6);
+        }
+#endif
+
+        time_t tt;
+    #if HAVE_GMTIME_R
+        struct tm ut;
+    #endif  /* HAVE_GMTIME_R */
+        struct tm *utPtr;
+        struct timeval timeval;
+        int timeofday_retval;
+
+        tt = time(NULL);
+
+        timeofday_retval = gettimeofday(&timeval,NULL);
+
+    #if HAVE_GMTIME_R
```

Comment:
The macro determine whether gmtime_r exist on the platform.
As @digitalinfinity pointed out earlier, we may be able to assume it exist?